BrainPortal/lib/cbrain_task_generators/templates/bourreau.rb.erb

```diff
@@ -299,10 +299,10 @@ class CbrainTask::<%= name %> < ClusterTask
 
         self.addlog("Saving result file #{path}")
         name = File.basename(path)
-
-        output = safe_userfile_find_or_new((
-          File.directory?(path) ? FileCollection : Userfile.suggested_file_type(name)
-        ), :name => name)
+        
+	userfile_class = Userfile.suggested_file_type(name)
+        userfile_class ||= ( File.directory?(path) ? FileCollection : SingleFile )
+        output = safe_userfile_find_or_new(userfile_class, :name => name)
```

Comment:
Nitpick; it seems like the indentation/whitespace is kinda wonky; line 302 has EOL whitespace, 303 has incorrect indentation and 304 has extra whitespace around the parentheses for the ternary operator (unlike anywhere else in the template). Shouldn't it look like:
```ruby
userfile_class   = Userfile.suggested_file_type(name)
userfile_class ||= (File.directory?(path) ? FileCollection : SingleFile)
output = safe_userfile_find_or_new(userfile_class, :name => name)
```